### ISSUE TITLE
FA Animated icons + custom icons

### ISSUE THREAD

**Issue by hmontes** [gorangajic/react-icons#105]
Hi.

1. According to FA website: http://fontawesome.io/examples/ the icons can be animated. This is implemented in your package?

2. Can i add custom icons? Like this popular library? https://github.com/thesabbir/simple-line-icons/

Thank you

**Comment by kumarharsh** [comment]
To add 'animated' icons, just add a css class on the JSX: `<IconSpinner className="spin" />`

In my case, I'm adding custom icons by making the svg in Illustrator/Affinity, compressing it via [SVGOMG](https://jakearchibald.github.io/svgomg/), and then creating my own icon file, say 'Bulb.js', with the following content:

```
import React from 'react';
import Icon from 'react-icon-base';

const IconBulb = (props) => (
    <Icon viewBox="0 0 40 40" {...props}>
        {my icon svg}
    </Icon>
);

export default IconBulb;
```

**Comment by hmontes** [comment]
```<FaCircleONotch className="spin" />```

Don't work :(

**Comment by marcellodesales** [comment]
I'm also looking for the `spin` or `bounce` of icons

**Comment by gtournie** [comment]
@hmontes: the class is 'fa-spin' and not 'spin'

`<FaCircleONotch className="fa-spin" />` should work

**Comment by Shadowstep33** [comment]
I don't believe this package pulls in the CSS so I'm of the opinion you'll have to define your own fa-spin class

**Comment by wallaceturner** [comment]
adding `className="fa-spin"` did not work so I added the fa-spin to my own CSS

```
.fa-spin {
  animation: fa-spin 2s infinite linear;
}
@keyframes fa-spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(359deg);
  }
}
```

**Comment by daymansiege** [comment]
You can also install font-awesome and use it's CSS
```
npm install --save font-awesome
```

```
import 'font-awesome/css/font-awesome.min.css';
```

**Comment by Vladis466** [comment]
For those of you who just want some pasta. Feel free to remove the css classes. I use tailwindcss. Also this is typescript.

`
import React from "react";

const Spinner = (props: { size?: string, color?: string }) => {
  const { size = 'md', color = 'black' } = props;
  const width: {[key:string]: string}= {
    sm: "w-6",
    md: "w-8",
    lg: "w-10",
  };

  return (
    <div className=" px-5 py-4">
      <div className={`text-center ${width[size] || 'md'}`}>
        <svg className="animate-spin" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
          <circle style={{  
            display: 'block', 
          fill: 'transparent', 
          stroke: `${color}`, 
          strokeLinecap: 'round', 
          strokeDasharray: 483, 
          strokeDashoffset: 280,
          strokeWidth: '10px',
          transformOrigin: '50% 50%'}} 
          cx="50" cy="50" r="45"
        />
        </svg>
      </div>
    </div>
  );
};

export default Spinner;
`

**Comment by francijpg** [comment]
Using [react-icons](https://react-icons.github.io/react-icons/), you can add the classes `"fas fa-spin"` and get the spinner effect on your icon.

`
import { FaGithub } from "react-icons/fa"`

`
<FaGithub className="fas fa-spin"></FaGithub>
`
